consensus/src/consensus/mod.rs

```diff
@@ -842,10 +870,73 @@ impl ConsensusApi for Consensus {
         })
     }
 
+    fn get_block_even_if_header_only(&self, hash: Hash) -> ConsensusResult<Block> {
+        let Some(status) = self.statuses_store.read().get(hash).unwrap_option().filter(|&status| !status.has_block_header()) else {
+            return Err(ConsensusError::BlockNotFound(hash));
+        };
+        Ok(Block {
+            header: self.headers_store.get_header(hash).unwrap(),
+            transactions: if status.is_header_only() { Arc::new(vec![]) } else { self.block_transactions_store.get(hash).unwrap() },
+        })
+    }
+
+    fn get_block_info(&self, hash: Hash) -> ConsensusResult<BlockInfo> {
+        let (exists, block_status) = match self.get_block_status(hash) {
+            Some(status) => (true, status),
+            None => (false, BlockStatus::StatusInvalid),
+        };
+
+        // If the status is invalid, then we don't have the necessary reachability data to check if it's in PruningPoint.Future.
+        if block_status == BlockStatus::StatusInvalid {
+            return Ok(BlockInfo::with_exists(exists));
+        }
+
+        // FIXME: check this call
+        let ghostdag_data = self.ghostdag_store.get_data(hash).unwrap();
+
+        Ok(BlockInfo::new(
+            exists,
+            block_status,
+            ghostdag_data.blue_score,
+            ghostdag_data.blue_work,
+            ghostdag_data.selected_parent,
+            (*ghostdag_data.mergeset_blues).clone(),
+            (*ghostdag_data.mergeset_reds).clone(),
+        ))
+    }
+
+    fn get_block_children(&self, hash: Hash) -> Option<Arc<Vec<Hash>>> {
+        self.relations_stores.read()[0].get_children(hash).unwrap_option()
+    }
+
+    fn get_block_parents(&self, hash: Hash) -> Option<Arc<Vec<Hash>>> {
+        self.relations_stores.read()[0].get_parents(hash).unwrap_option()
+    }
+
     fn get_block_status(&self, hash: Hash) -> Option<BlockStatus> {
         self.statuses_store.read().get(hash).unwrap_option()
     }
 
+    fn get_block_acceptance_data(&self, hash: Hash) -> ConsensusResult<Arc<AcceptanceData>> {
+        self.validate_block_exists(hash)?;
+        Ok(self.acceptance_data_store.get(hash).unwrap())
+    }
+
+    fn get_blocks_acceptance_data(&self, hashes: &[Hash]) -> ConsensusResult<Vec<Arc<AcceptanceData>>> {
+        hashes
+            .iter()
+            .copied()
+            .map(|hash| {
+                self.validate_block_exists(hash)?;
+                Ok(self.acceptance_data_store.get(hash).unwrap())
```

Comment:
the logic here relies on the fact that `hashes` are chain blocks obtained by a call to `get_virtual_chain` (otherwise not all blocks have acceptance data). Given this fact I think we can skip the `validate_block_exists` part and add a doc over `get_blocks_acceptance_data` def in `ConsensusApi` explaining this assumption. Same goes for `get_block_acceptance_data`. 
Alternatively, do not unwrap on `self.acceptance_data_store.get(hash)` and propagate that error instead.